addons/account/tests/test_payment_term.py

```diff
@@ -397,3 +397,25 @@ def assert_payment_term_values(expected_values_list):
                     'discount_amount_currency': 362.30,
                 },
             ])
+
+    @users('__system__')
+    def test_payment_term_multi_company(self):
+        """
+        Ensure that the payment term is determined by `move.company_id` rather than `user.company_id`.
+        OdooBot has `res.company(1)` set as the default company. The test checks that the payment term correctly reflects
+        the company associated with the move, independent of the user's default company.
+        """
+        company_1, company_2 = self.company_data_2.get('company'), self.company_data.get('company')
```

Comment:
nitpick, but it's a little little confusing with the order and naming of assignment here, do you think you could swap them?

Reply:
I did this for debugging purpose. 

Didn't investigate but company_date_2 uses a company created before. So it's a bit confusing when debugging to know which company we are looking at